### ISSUE TITLE
TextField disabled with suffix giving failed instances of color-contrast in Accessibility Insights for Web

### ISSUE THREAD

**Issue by v-vibsingh** [microsoft/fluentui#16287]
### Environment Information

- **Package version(s)**: "office-ui-fabric-react": "7.153.2", "office-ui-fabric-core": "11.0.0",
- **Browser and OS versions**: Chrome and Edge both with Windows 10 20H2

### Describe the issue:

TextField disabled with suffix giving failed instances of `color-contrast` in `Accessibility Insights for Web`.

`Element has insufficient color contrast of 2.35 (foreground color: #a19f9d, background color: #f3f2f1, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1`

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/v-vibsingh/pen/dypVPGe

#### Actual behavior:

`Accessibility Insights for Web` showing failed instances as `**color-contrast** Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds`.

#### Expected behavior:

`Accessibility Insights for Web` should show zero failed instances.

### Documentation describing actual behavior

https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield

![image](https://user-images.githubusercontent.com/76475047/102888642-32dfdf80-447f-11eb-921b-8144ea083cf5.png)

**Comment by v-vibsingh** [comment]
Hi Team,

Please provide update for this issue.

Regards,
Vibhas

**Comment by ecraig12345** [comment]
Disabled controls are generally exempt from color contrast requirements, so I'd assume that is also the case here. @smhigley can you confirm whether that's correct? If so, I'll need to do some more investigation to figure out how to keep the prefix and suffix from being flagged by A11y Insights (they're a bit of a special case since they're not a direct part of the input field).

**Comment by v-vibsingh** [comment]
ecraig12345,
Any updates on this issue.

**Comment by smhigley** [comment]
Elizabeth is right, disabled user interface elements are not required to meet the minimum color contrast ratio in WCAG. Accessibility Insights is flagging the prefix text in that example because it's semantically out of the `<input disabled>` element itself, but it is conceptually part of it.

In general, there's a lot of discussion in the accessibility community on how to best represent disabled controls and text within them, since low contrast obviously disproportionately impacts people with low vision. The problem with just increasing contrast is that then the control can look like it's not disabled, which is also an issue. There have been some other options proposed, like using visible text with a strikethrough, but I'm not a designer so I can't really comment on them. @betrue-final-final probably has more informed opinions :).

This isn't a compliance issue, but I'll leave this open for now for more conversation on design and color options. Thanks for raising it!
